modules/utxo-lib/test/testutil/tnumber.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * Multiply a decimal amount (optionally to a bigint)
+ * Throws error if resulting value is not a safe integer number
+ * @param value - decimal amount
+ * @param amountType - desired output type
+ * @return value * 1e8, as amountType
+ */
+export function decimalToSats<TNumber extends number | bigint>(
+  value: number,
+  amountType: 'number' | 'bigint' = 'number'
+): TNumber {
+  if (amountType === 'number') {
+    const scaledValue = value * 1e8;
+    if (!Number.isSafeInteger(scaledValue)) {
+      throw new Error('input value cannot be scaled to safe integer number');
+    }
+    return scaledValue as TNumber;
+  } else {
+    const [integerString, decimalString] = value.toFixed(8).split('.');
+    return (BigInt(integerString) * BigInt(1e8) + BigInt(decimalString)) as TNumber;
```

Comment:
It looks like this cast isn't needed? To be sure I checked out the branch and built after removing this and it worked

Reply:
Same for the cast on line 17.

Reply:
This is test code, build will work, but tests will fail without the casts.

Reply:
Really? The casts don't affect runtime, so execution should be the same. Are the tests using a different tsconfig or something such that ts-node is failing?

Reply:
The tests are run as typescript, so aren't compiled until they are executed.